war/src/main/webapp/scripts/hudson-behavior.js

```diff
@@ -2923,18 +2923,24 @@ function applyErrorMessage(elt, rsp) {
     if (rsp.status == 200) {
         elt.innerHTML = rsp.responseText;
     } else {
-        var id = 'valerr' + (iota++);
-        elt.innerHTML = '<a href="" onclick="document.getElementById(\'' + id
-        + '\').style.display=\'block\';return false">ERROR</a><div id="'
-        + id + '" style="display:none">' + rsp.responseText + '</div>';
-        var error = document.getElementById('error-description'); // cf. oops.jelly
-        if (error) {
-            var div = document.getElementById(id);
-            while (div.firstElementChild) {
-                div.removeChild(div.firstElementChild);
-            }
-            div.appendChild(error);
-        }
+        var parser = new DOMParser();
+        var doc = parser.parseFromString(rsp.responseText, 'text/html');
+        var message = doc.querySelector('#error-description') || doc.body;
+        // Remove styles and scripts
+        var harmfulElements = doc.querySelectorAll('style, script');
+        harmfulElements.forEach(function(el) {
+            el.parentElement.removeChild(el);
+        });
+        // Clear out previous errors
+        while(elt.firstChild) {
+            elt.removeChild(elt.firstChild);
+        }
+        var details = document.createElement('details');
+        var summary = document.createElement('summary');
+        summary.textContent = 'Field Validation ERROR';
+        details.appendChild(summary);
+        details.appendChild(document.adoptNode(message));
+        elt.appendChild(details);
```

Comment:
This is how it looks:

<img width="936" alt="image" src="https://user-images.githubusercontent.com/47652/124082031-8d6c8080-da4c-11eb-9473-19014c82fe35.png">


And opened:

<img width="933" alt="image" src="https://user-images.githubusercontent.com/47652/124082110-9eb58d00-da4c-11eb-8485-b3a6089395b5.png">